src/Views/SilenceForm/Views.elm

```diff
@@ -77,30 +39,65 @@ silenceForm kind silence =
                         , label [ class "f6 dib mb2 mr2 w-40" ] [ text "Name" ]
                         , label [ class "f6 dib mb2 mr2 w-40" ] [ text "Value" ]
                         ]
-                    , div [] (List.map boundMatcherForm silence.matchers)
-                    , iconButtonMsg "blue" "fa-plus" (AddMatcher silence)
-                    , formField "Creator" silence.createdBy (UpdateCreatedBy silence)
-                    , textField "Comment" silence.comment (UpdateComment silence)
+                    , div [] (List.indexedMap matcherForm form.matchers)
+                    , iconButtonMsg "blue" "fa-plus" (AddMatcher |> UpdateField)
+                    , formField "Creator" form.createdBy (UpdateCreatedBy >> UpdateField)
+                    , textField "Comment" form.comment (UpdateComment >> UpdateField)
                     , div [ class "mt3" ]
-                        [ a [ class "f6 link br2 ba ph3 pv2 mr2 dib blue", onClick (CreateSilence silence) ] [ text "Create" ]
-                          -- Reset isn't working for "New" -- just updates time.
-                        , a [ class "f6 link br2 ba ph3 pv2 mr2 dib dark-red", href url ] [ text "Reset" ]
+                        [ createSilence silence
+                        , a
+                            [ class "f6 link br2 ba ph3 pv2 mr2 dib dark-red"
+                            , onClick resetClick
+                            ]
+                            [ text "Reset" ]
                         ]
                     ]
-                    |> (Html.map MsgForSilenceForm)
-                , div [ class "mt3" ]
-                    [ a [ class "f6 link br2 ba ph3 pv2 mr2 dib dark-green", onClick <| PreviewSilence silence ] [ text "Show Affected Alerts" ]
+                , preview silence
+                ]
+            ]
+
+
+createSilence : Result String Silence -> Html SilenceFormMsg
+createSilence silenceResult =
+    case silenceResult of
+        Ok silence ->
+            a
+                [ class "f6 link br2 ba ph3 pv2 mr2 dib blue"
+                , onClick (CreateSilence silence)
```

Comment:
Ou, my bad. Sorry. This should definitely go into the requirements for alertmanager api v2.